### PR TITLE
[BROWSEUI] Remove useless variable and unreachable code

### PR DESCRIPTION
**Pull request by SergeGautherie**
Addendum to 0c47416 (r72003).

JIRA issue: [CORE-12804](https://jira.reactos.org/browse/CORE-12804)

Co-Authored-By: Victor Martinez Calvo <...>

### CODE REVIEW

File: dll/win32/browseui/explorerband.cpp [hunk 1]
```diff
@@ -726,18 +726,16 @@ BOOL CExplorerBand::NavigateToPIDL(LPITEMIDLIST dest, HTREEITEM *item, BOOL bExp
...
     TVITEM                              tvItem;
 
     if (!item)
         return FALSE;
 
-    found = FALSE;
     current = m_hRoot;
     parent = NULL;
-    while(!found)
+    do
```
Comment (HBelusca) [inline dll/win32/browseui/explorerband.cpp:738]: please use `while (TRUE)` instead.

Comment (SergeGautherie) [reply dll/win32/browseui/explorerband.cpp]: Done.

**Comment by SergeGautherie** [comment]
> `Build / build-msvc-arm (14.1, arm, Debug) (pull_request) Failing`

The usual
`LINK : fatal error LNK1318: Unexpected PDB error; ACCESS_DENIED (15) 'D:\a\reactos\reactos\build\msvc_pdb\vfd.pdb'`

File: dll/win32/browseui/explorerband.cpp [hunk 2]
```diff
@@ -726,18 +726,16 @@ BOOL CExplorerBand::NavigateToPIDL(LPITEMIDLIST dest, HTREEITEM *item, BOOL bExp
...
     TVITEM                              tvItem;
 
     if (!item)
         return FALSE;
 
-    found = FALSE;
     current = m_hRoot;
     parent = NULL;
-    while(!found)
+    while(TRUE)
```
Comment (HBelusca) [inline dll/win32/browseui/explorerband.cpp:738]:
```suggestion
    while (TRUE)
```

**Review by HBelusca — APPROVED** [review]

File: dll/win32/browseui/explorerband.cpp (hunk 2 above)
Comment (SergeGautherie) [reply dll/win32/browseui/explorerband.cpp]: Added.

**Review by binarymaster — APPROVED** [review]

**Comment by binarymaster** [comment]
@SergeGautherie recently I started to check commit author fields that are being merged into master branch, and I noticed you have an autogenerated email address by GitHub.

You need to uncheck this option in your [GitHub email settings](https://github.com/settings/emails):

image

So when maintainers use "Squash and merge" button, the commit would end up merged with correct fields.

**Comment by SergeGautherie** [comment]
> Keep my email addresses private

As far as I understand, this option applies to all addresses, as in there is no way to have some public and some private,
so I want to keep it enabled: blame GitHub behavior!

**Comment by binarymaster** [comment]
When option is unchecked, it uses primary address.

**Comment by HBelusca** [comment]
> @SergeGautherie recently I started to check commit author fields that are being merged into master branch, and I noticed you have an autogenerated email address by GitHub.
> 
> You need to uncheck this option in your [GitHub email settings](https://github.com/settings/emails):
> 
> So when maintainers use "Squash and merge" button, the commit would end up merged with correct fields.

This option only applies to the **single** public email that has been chosen in the main settings:
image

**Comment by SergeGautherie** [comment]
> When option is unchecked, it uses primary address.

Yeah, that's how I understand it.

> This option only applies to the single public email that has been chosen in the main settings

I doubt that:
GitHub doesn't let me set a public address, even if I wanted to,
the option reads 'Keep my email addresses private', notice plural.

> it is now possible to select email in GitHub interface when doing Squash & Merge on a Pull Request

That feature is not available to me, so I can't comment on it.

Again, my ReactOS address is declared on GitHub and correctly set on my commits.
Note:
Some of my commits pushed to master have replaced author address and GitHub committer,
some others have correct addresses on both fields.
Why the two different behaviors? I have no idea.

**Comment by binarymaster** [comment]
> Why the two different behaviors? I have no idea.

GitHub interface was updated recently, and now it's clearly showing this thing on merging:

image

**Pull request merged**